openedx/core/djangoapps/user_api/accounts/api.py

```diff
@@ -61,34 +63,19 @@ def get_account_settings(requesting_user, username=None, configuration=None, vie
     if username is None:
         username = requesting_user.username
 
-    has_full_access = requesting_user.username == username or requesting_user.is_staff
-    return_all_fields = has_full_access and view != 'shared'
-
-    existing_user, existing_user_profile = _get_user_and_profile(username)
-
-    user_serializer = AccountUserSerializer(existing_user)
-    legacy_profile_serializer = AccountLegacyProfileSerializer(existing_user_profile)
-
-    account_settings = dict(user_serializer.data, **legacy_profile_serializer.data)
-
-    if return_all_fields:
-        return account_settings
-
-    if not configuration:
-        configuration = settings.ACCOUNT_VISIBILITY_CONFIGURATION
-
-    visible_settings = {}
-
-    profile_visibility = _get_profile_visibility(existing_user_profile, configuration)
-    if profile_visibility == ALL_USERS_VISIBILITY:
-        field_names = configuration.get('shareable_fields')
-    else:
-        field_names = configuration.get('public_fields')
+    try:
+        existing_user = User.objects.get(username=username)
+    except ObjectDoesNotExist:
+        raise UserNotFound()
 
-    for field_name in field_names:
-        visible_settings[field_name] = account_settings.get(field_name, None)
+    has_full_access = requesting_user.username == username or requesting_user.is_staff
+    admin_fields = has_full_access and view != 'shared'
```

Comment:
Name this differently to make it clear that this is a boolean (it isn't the list of admin fields).